File: backend/mr-db-sql/src/main/java/io/featurehub/db/publish/CacheBroadcast.java
package io.featurehub.db.publish;

import io.featurehub.dacha.model.PublishEnvironment;
import io.featurehub.dacha.model.PublishFeatureValue;
import io.featurehub.dacha.model.PublishServiceAccount;

public interface CacheBroadcast {
  void publishEnvironment(PublishEnvironment eci);

  void publishServiceAccount(PublishServiceAccount saci);

  void publishFeature(PublishFeatureValue feature);
}

File: backend/mr-eventing-common/src/main/kotlin/io/featurehub/mr/events/common/CacheBroadcast.kt
package io.featurehub.mr.events.common

import io.featurehub.dacha.model.PublishEnvironment
import io.featurehub.dacha.model.PublishFeatureValues
import io.featurehub.dacha.model.PublishServiceAccount

interface CacheBroadcast {
    fun publishEnvironment(eci: PublishEnvironment)
    fun publishServiceAccount(saci: PublishServiceAccount)
    fun publishFeatures(features: PublishFeatureValues)
}
